### PR TITLE
[Review] Rename component tag to parameter block

### PR DESCRIPTION
**Pull request by tcahuzax**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150560147%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150564250%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150607824%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150612004%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881296%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881309%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881336%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881414%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150624662%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150624687%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/295%23issuecomment-150560147%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%5Cr%5Cn%5Cr%5Cn%40dawagner%20%40krocard%20Plz%20review%22%2C%20%22created_at%22%3A%20%222015-10-23T12%3A38%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22do%20not%20review%3A%20will%20be%20reworked%22%2C%20%22created_at%22%3A%20%222015-10-23T13%3A02%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A27%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20only%20have%20one%20minor%20comment%20for%20a%20%5C%22line%20too%20long%5C%22%20issue%20that%20you%20did%20not%20even%20introduce%20yourself.%20If%20there%20isn%27t%20any%20comment%20from%20%40krocard%20I%27ll%20just%20merge.%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A42%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A1%3A%22%2C%20%22created_at%22%3A%20%222015-10-23T16%3A23%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-23T16%3A23%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%208c524421731dd337aa2bc14865a3d2ef67d68ad0%20parameter/ConfigurableElement.cpp%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881296%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60auto%60%20%3D%3E%20%60bool%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A52%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableElement.cpp%3AL74-98%22%7D%2C%20%22Pull%208c524421731dd337aa2bc14865a3d2ef67d68ad0%20test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881336%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22miss%20alignment.%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A53%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml%3AL23-31%22%7D%2C%20%22Pull%208c524421731dd337aa2bc14865a3d2ef67d68ad0%20parameter/DomainConfiguration.cpp%2018%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/295%23discussion_r42881309%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60auto%60%20%3D%3E%20%60bool%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A52%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%60auto%60%20%3D%3E%20%60bool%60%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-23T15%3A53%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.cpp%3AL146-177%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 8c524421731dd337aa2bc14865a3d2ef67d68ad0 parameter/ConfigurableElement.cpp 17'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/295#discussion_r42881296'>File: parameter/ConfigurableElement.cpp:L74-98</a></b>
- <a href='https://github.com/krocard'></a> `auto` => `bool` ?
- [ ] <a href='#crh-comment-Pull 8c524421731dd337aa2bc14865a3d2ef67d68ad0 parameter/DomainConfiguration.cpp 18'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/295#discussion_r42881309'>File: parameter/DomainConfiguration.cpp:L146-177</a></b>
- <a href='https://github.com/krocard'></a> `auto` => `bool` ?
- <a href='https://github.com/krocard'></a> `auto` => `bool` ?
- [ ] <a href='#crh-comment-Pull 8c524421731dd337aa2bc14865a3d2ef67d68ad0 test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/295#discussion_r42881336'>File: test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml:L23-31</a></b>
- <a href='https://github.com/krocard'></a> miss alignment.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/295#issuecomment-150560147'>General Comment</a></b>
- <a href='https://github.com/tcahuzax'></a> :+1:
@dawagner @krocard Plz review
- <a href='https://github.com/tcahuzax'></a> do not review: will be reworked
- <a href='https://github.com/tcahuzax'></a> :+1:
- <a href='https://github.com/dawagner'></a> I only have one minor comment for a "line too long" issue that you did not even introduce yourself. If there isn't any comment from @krocard I'll just merge.
- <a href='https://github.com/tcahuzax'></a> :1:

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/295?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/295?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/295'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by tcahuzax** [comment]
:+1:

@dawagner @krocard Plz review

**Comment by tcahuzax** [comment]
do not review: will be reworked

**Comment by tcahuzax** [comment]
:+1:

**Comment by dawagner** [comment]
I only have one minor comment for a "line too long" issue that you did not even introduce yourself. If there isn't any comment from @krocard I'll just merge.

File: parameter/ConfigurableElement.cpp [hunk 1]
```diff
@@ -74,12 +74,24 @@ bool CConfigurableElement::serializeXmlSettings(CXmlElement& xmlConfigurationSet
...
-
-                return false;
+            if (xmlChildConfigurableElementSettingsElement.getType() != pChildConfigurableElement->getXmlElementName()) {
+
+                // "Component" tag has been renamed to "ParameterBlock", but retro-compatibility shall
+                // be ensured.
+                //
+                // So checking if this case occurs, i.e. element name is "ParameterBlock"
+                // but xml setting name is "Component".
+                auto compatibilityCase =
```
Comment (krocard) [inline parameter/ConfigurableElement.cpp]: `auto` => `bool` ?

File: parameter/DomainConfiguration.cpp [hunk 2]
```diff
@@ -146,16 +146,31 @@ bool CDomainConfiguration::importOneConfigurableElementSettings(
...
-        context.setError(
-                "Couldn't find settings for " +
-                destination->getKind() + " " + destination->getName() +
-                " for Configuration " + getPath());
+        // "Component" tag has been renamed to "ParameterBlock", but retro-compatibility shall
+        // be ensured.
+        //
+        // So checking if this case occurs, i.e. element name is "ParameterBlock"
+        // but found xml setting name is "Component".
+        auto compatibilityCase =
```
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: `auto` => `bool` ?

File: test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml [hunk 3]
```diff
@@ -23,6 +23,8 @@
 		<Settings>
 			<Configuration Name="Conf_1_1">
 				<ConfigurableElement Path="/Test/Test/TEST_DIR">
+                    <!-- The 'Component' tag has been renamed to 'ParameterBlock', but retro-compatibility is needed.
+                         So keeping 'Component' to test it. -->
```
Comment (krocard) [inline test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml]: miss alignment.

File: parameter/DomainConfiguration.cpp (hunk 2 above)
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: `auto` => `bool` ?

**Comment by tcahuzax** [comment]
:1:

**Comment by tcahuzax** [comment]
:+1:

File: parameter/ConfigurableElement.cpp (hunk 1 above)
Comment (krocard) [inline parameter/ConfigurableElement.cpp]: :+1:<a href='#crh-update-none'></a>

File: test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml (hunk 3 above)
Comment (krocard) [inline test/functional-tests-legacy/xml/XML_Test/Reference_Compliant.xml]: :+1:<a href='#crh-update-none'></a>

File: parameter/DomainConfiguration.cpp (hunk 2 above)
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**